draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1062,98 +923,338 @@ trace has a different value for a given field, this field MUST NOT be added to
-separate, optional, field named "raw" of type RawInfo. The exact definition of
-entities, headers, trailers and payloads depend on the protocol used.
-
-~~~ cddl
-RawInfo = {
-
-    ; the full byte length of the entity (e.g., packet or frame),
-    ; including possible headers and trailers
-    ? length: uint64
+# Concrete Event Types and Event Schema {#event-types-and-schema}
 
-    ; the byte length of the entity's payload,
-    ; excluding possible headers or trailers
-    ? payload_length: uint64
+Concrete event types are defined in event schema, which declare a namespace
+consisting of one or more categories, each containing related event types. This
+document defines the `main` event schema. Other examples are the `quic`
+{{QLOG-QUIC}} and `h3` {{QLOG-H3}} event schema.
 
-    ; the (potentially truncated) contents of the full entity,
-    ; including headers and possibly trailers
-    ? data: hexstring
-}
-~~~
-{: #raw-info-def title="RawInfo definition"}
+Concrete events MAY extend any part of the abstract Event class, including extending the "data" field ({{data-field}}) of adding custom fields.
 
-The RawInfo:data field can be truncated for privacy or security purposes, see
-{{truncated-values}}. In this case, the length and payload_length fields should
-still indicate the non-truncated lengths when used for debugging purposes.
+Event schema MUST define a registered non-empty namespace of type `text`.
 
-This document does not specify explicit header_length or trailer_length fields.
-In protocols without trailers, header_length can be calculated by subtracting
-the payload_length from the length. In protocols with trailers (e.g., QUIC's
-AEAD tag), event definition documents SHOULD define how to support header_length
-calculation.
+Event categories MUST belong to a single event schema. The MUST have a
+registered non-empty globally-unique identifier of type `text` and MUST have a
+single dereferencable URI. That URI MUST be absolute and MUST indicate the
+category identifier using a fragment identifier (characters after a "#" in the
+URI). For event categories in schema defined by RFCs, the URI SHOULD be a URN of
+the form `urn:ietf:params:qlog:<schema namespace>#<category identifier>`.
+Private or non-standard event categories can use other URI formats. URIs that
+contain a domain name SHOULD also contain a month-date in the form mmyyyy. The
+definition of the category and assignment of the URI MUST have been authorized by
+the owner of the domain name on or very close to that date. (This avoids
+problems when domain names change ownership.)
 
-# Common events and data classes
+The registration requirements for extension schema URIs are detailed in
+{{iana}}.
 
-There are some event types and data classes that are common across protocols,
-applications, and use cases. This section specifies such common definitions.
+Concrete event types MUST belong to a single event category and MUST have a
+non-empty identifier of type `text`.
```

Comment:
```suggestion
non-empty name of type `text`.
```

since you later use "event type identifier" to indicate just the "type name" (second part of the `name`).